search/tutorials/tutorial_ordering.md

```diff
@@ -1,136 +1,216 @@
+    ```bash
+    pip install virtualenv
+    virtualenv myenv
+    source myenv/bin/activate
+    ```
+1. Next, install Google packages:
+    ```bash
+    pip install google
+    pip install google-cloud-retail
+    ```
+
+## Clone the Retail code samples (optional)
+
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
+
+Clone the Git repository with all the code samples to learn the Retail features and check them in action.
+
+<!-- TODO(ianan): change the repository link -->
+1. Run the following command in the Terminal:
+    ```bash
+    git clone https://github.com/t-karasova/grs-samples-python.git
+    ```
 
-As you can see now, ```results[]``` are ordered by price descending.
+    The code samples for each of the Retail services are stored in different directories.
 
-## Ordering by a single field. Product sorting
+1. Go to the ```grs-samples-python``` directory. It's our starting point to run more commands.
+    ```bash
+    cd grs-samples-python
+    ```
 
-Next, let's change the ordering direction to show the cheapest products first.
+## Import catalog data (optional)
 
-To do that, find the comment 
+This step is required if this is the first Retail API Tutorial you run.
+Otherwise, you can skip it.
 
-"#TRY DIFFERENT ORDERING EXPRESSIONS HERE:" 
+### Upload catalog data to Cloud Storage
 
-and change the ordering expression to the following one:
+We have prepared a JSON file with valid products in the `search` directory:
 
-```order = 'price asc'``` or just ```order = 'price'``` - those are equivalent expressions since ascending is the default ordering direction.
+**search/products_for_search.json**
 
-Run the sample in a terminal using the command:
+You can use this file in the tutorial.
+
+1.  In your own Google Cloud project, go to Cloud Storage.
+1.  Click **Create bucket**, give your Project ID in the name, and
+    click **Create**.
+1.  From the Cloud Shell Terminal, run the following command:
+    ```bash
+    gsutil cp search/products_for_search.json gs://<YOUR_PROJECT_ID>
+    ```
+
+1.  Now you can see that the file is uploaded to the Cloud Storage bucket.
+
+### Import products to the Retail Catalog
+
+To import the prepared products to a catalog, run the following command in the Terminal:
 
 ```bash
-python search_with_ordering.py
+python product/import_products_gcs.py
 ```
 
-You have sorted the products by price ascending.
+## Order by a single field: ordering expression
+
+To use the ordering feature, you need to specify the field and the ordering direction. You can order by both the text and numeric fields.
+
+1. Open <walkthrough-editor-select-regex filePath="cloudshell_open/grs-samples-python/search/search_with_ordering.py" regex="TRY DIFFERENT ORDERING EXPRESSIONS HERE">search_with_ordering.py</walkthrough-editor-select-regex>.
+
+1. Order the search results by price when more expensive items come first. To do that, set the ordering expression as follows:
```

Comment:
Order the search results by price to list more expensive items first. To do that, set the ordering expression as follows: